plug-ins/skills_impl/transportspell.cpp

```diff
@@ -319,65 +344,88 @@ bool SummonMovement::checkCaster( )
+    }
 
     return true;
 }    
 
 bool SummonMovement::checkVictim( )
 {
-    if (ch == caster)
+    if (ch == caster) {
+        caster->pecho( "Ты с глубоким удовлетворением призываешь сам{Smого{Sfу{Sx себя." );
+        caster->pecho( "К немедленному ответу за бесцельно прожитые годы." );      
         return false;
+    }
     
-    if (ch->is_immortal( ) && !caster->is_immortal( ))
+    if (ch->is_immortal( ) && !caster->is_immortal( )) {
+        caster->pecho( "Призывать Бессмертных лучше другими способами." );
         return false;
+    }
 
-    if (IS_SET(ch->in_room->room_flags, ROOM_SAFE|ROOM_NOSUMMON))
+    if (IS_SET(ch->in_room->room_flags, ROOM_SAFE|ROOM_NOSUMMON)) {
+        caster->pecho( "Местность, в которой находится твоя жертва, защищена Богами." );
         return false;
+    }
 
-    if (ch->fighting != 0)
+    if (ch->fighting != 0) {
+        caster->pecho( "Разум твоей жертвы в смятении, тебе не удается установить ментальную связь." );
         return false;
+    }
 
-    if (ch->isAffected( gsn_mental_block ))
+    if (ch->isAffected( gsn_mental_block )) {
+        caster->pecho( "Непроницаемый экран защищает твою жертву, тебе не удается установить ментальную связь." );
         return false;
+    }
 
-    if (ch->isAffected( gsn_spellbane ))
+    if (ch->isAffected( gsn_spellbane )) {
+        caster->pecho( "Ненависть к магии защищает твою жертву, тебе не удается установить ментальную связь." );
         return false;
+    }
 
-    if (IS_SET(ch->imm_flags, IMM_SUMMON))
+    if (IS_SET(ch->imm_flags, IMM_SUMMON)) {
+        caster->pecho( "Непроницаемый экран защищает твою жертву, тебе не удается установить ментальную связь." );
         return false;
-        
+    }
+  
     if (ch->is_npc( )) {
-        if (IS_SET(ch->act,ACT_AGGRESSIVE))
+        if (IS_SET(ch->act,ACT_AGGRESSIVE) || IS_AFFECTED(ch,AFF_BLOODTHIRST)) {
+            caster->pecho( "Призывать агрессивных существ запрещено Богами." );
             return false;
+        }
             
-        if (ch->getNPC( )->behavior
-            && IS_SET(ch->getNPC( )->behavior->getOccupation( ), (1 << OCC_SHOPPER)))
-            return false;
-
-        if (is_safe_nomessage( caster, ch ))
+        if (ch->getNPC( )->behavior && IS_SET(ch->getNPC( )->behavior->getOccupation( ), (1 << OCC_SHOPPER))) {
+            caster->pecho( "Призывать продавцов запрещено Богами." );
             return false;
+        }
+            
+        if (is_safe( caster, ch ))
+            return false;            
     }
-    else {
-        if (!IS_SET(ch->act,PLR_NOSUMMON))
-            return true;
-        
-        if (!is_safe_nomessage( caster, ch ))
-            return true;
-        
-        if (!caster->is_npc( )) {
-            if (ch->getClan( )->isDispersed( ))
-                return false;
-
-            if (caster->getClan( ) != ch->getClan( ))
+    else {      
+        if (!caster->is_npc( )) {      
+            if (is_safe( caster, ch ))
```

Comment:
@fiorine New logic will prevent players from the same clan from summoning each other if they're not in PK range. Also, `isDispersed` will return `true` for outsiders and exiles. Easier to compare with `clan_none`:  `if (ch->getClan() == clan_none)`...

Reply:
@ruffinakoza fixed